src/js/components/OptionalEnviromentComponent.jsx

```diff
@@ -14,6 +14,7 @@ var OptionalEnvironmentComponent = React.createClass({
 
   propTypes: {
     errorIndices: React.PropTypes.array,
+    generalError: React.PropTypes.string,
```

Comment:
Scrap that. I just realised we are overriding the getErrorMessage to return both clientside and serverside errors.